### PR TITLE
Add agendaEventMinHeight option to prevent short events overlapping

### PR DESCRIPTION
**Pull request by Stafie**
Issue: https://github.com/fullcalendar/fullcalendar/issues/961

Fiddle illustrating the problem: http://jsfiddle.net/jc1zx2q6/5/
Fiddle illustrating the solution: https://plnkr.co/edit/t5eAi1raDw4ZexwCrWy6?p=preview

### CODE REVIEW

**Comment by arshaw** [comment]
This is a good stopgap solution, using a JS setting, but I would prefer if the CSS was able to specify the min-height. In other words, I'd prefer if the DOM was consulted for the min-height.

I might need to merge this tho, to get it out there faster. Could you rename the setting to `agendaEventMinHeight` ?

File: src/agenda/TimeGrid.js [hunk 1]
```diff
@@ -477,7 +477,12 @@ var TimeGrid = FC.TimeGrid = InteractiveDateComponent.extend(StandardInteraction
 			dayDate = this.dayDates[seg.dayIndex];
 
 			seg.top = this.computeDateTop(seg.startMs, dayDate);
-			seg.bottom = this.computeDateTop(seg.endMs, dayDate);
+			if(this.computeDateTop(seg.endMs, dayDate) - seg.top < this.view.opt('minSegmentHeight')) {
```
Comment (arshaw) [inline src/agenda/TimeGrid.js]: could you cache the value of the `opt()` call in a variable at the top of the function?

File: src/agenda/TimeGridEventRenderer.js [hunk 2]
```diff
@@ -273,6 +273,7 @@ var TimeGridEventRenderer = EventRenderer.extend({
 		props.zIndex = seg.level + 1; // convert from 0-base to 1-based
 		props.left = left * 100 + '%';
 		props.right = right * 100 + '%';
+		props.minWidth = 'calc(' + (forwardCoord - backwardCoord) * 100 + '%' + ' - 2px)';
```
Comment (arshaw) [inline src/agenda/TimeGridEventRenderer.js]: what is this doing?

Comment (Stafie) [reply src/agenda/TimeGridEventRenderer.js]: I remember having some issues with events that had short width, but I couldn't replicate it now so I removed this.

File: src/agenda/TimeGrid.js (hunk 1 above)
Comment (Stafie) [reply src/agenda/TimeGrid.js]: done.

**Comment by Stafie** [comment]
I apologize for the delay, I made the adjustments you asked for.

File: src/agenda/agenda.css [hunk 3]
```diff
@@ -217,6 +217,11 @@ be a descendant of the grid when it is being dragged.
 
 /* short mode, where time and title are on the same line */
 
+.fc-time-grid-event.fc-short:hover {
```
Comment (arshaw) [inline src/agenda/agenda.css]: why was this added? if it's not related to the main feature in this PR, could you please remove it?

File: src/agenda/TimeGrid.js [hunk 4]
```diff
@@ -471,13 +471,19 @@ var TimeGrid = FC.TimeGrid = InteractiveDateComponent.extend(StandardInteraction
...
 		var dayDate;
+		var agendaEventMinHeight = this.view.opt('agendaEventMinHeight');
 
 		for (i = 0; i < segs.length; i++) {
 			seg = segs[i];
 			dayDate = this.dayDates[seg.dayIndex];
 
 			seg.top = this.computeDateTop(seg.startMs, dayDate);
-			seg.bottom = this.computeDateTop(seg.endMs, dayDate);
+			if(this.computeDateTop(seg.endMs, dayDate) - seg.top < this.view.opt('agendaEventMinHeight')) {
```
Comment (arshaw) [inline src/agenda/TimeGrid.js]: please us the `agendaEventMinHeight` you created above. also, you are calling `computeDateTop` twice. could you also please put that in a variable.

Also, please do this without an `if` statement. That should be possible with `Math.max`

Comment (Stafie) [reply src/agenda/TimeGrid.js]: Made the changes here.
Since we got rid of the `if` statement, now we use the `computeDateTop` once, so didn't have to put it in a variable.

File: src/agenda/agenda.css (hunk 3 above)
Comment (Stafie) [reply src/agenda/agenda.css]: This allows to view the event data when they are crammed together on the same line. GIF:
https://www.dropbox.com/s/b512d3xweimrwz6/event%20hovering.gif?dl=0

Let me know if you want to get rid of it, and I'll do a fixup.

Comment (arshaw) [reply src/agenda/agenda.css]: yes, could you get rid of it and put it into a different PR. it's a distinct enough feature, and up for debate as to what the exact API should be, and if it should be the new default.

Comment (Stafie) [reply src/agenda/agenda.css]: Got rid of it.

**Comment by arshaw** [comment]
thanks

**Comment by arshaw** [comment]
queued for release. i did a fix or two

**Comment by arshaw** [comment]
Released in [v3.6.0](https://github.com/fullcalendar/fullcalendar/releases/tag/v3.6.0)

docs: https://fullcalendar.io/docs/agenda/agendaEventMinHeight/
